### PR TITLE
[Simulator] Add compass module to simulator

### PR DESCRIPTION
**Pull request by microbit-robert**
Also tweaks the size of the sub-heading text for buttons and pins.

### CODE REVIEW

**Comment by microbit-matt-hillsdon** [comment]
image

Vertical spacing, particularly the differing space under the headings seems a bit off.

File: src/simulator/RangeSensor.tsx [hunk 1]
```diff
@@ -54,12 +54,17 @@ const RangeSensor = ({
     },
     [isFocused]
   );
+  const valuePercent = useMemo(
+    () => ((value - min) / (max - min)) * 100,
+    [min, max, value]
+  );
```
Comment (microbit-matt-hillsdon) [inline src/simulator/RangeSensor.tsx:60]: I suspect this calculation is very low cost compared with the bookkeeping around useMemo. Can it just be a variable calculated each render?

File: src/simulator/RangeSensor.tsx [hunk 2]
```diff
@@ -120,8 +125,15 @@ const RangeSensor = ({
...
               textAlign="center"
               mt="-8"
-              ml={-valueText.length / 2 + "ch"}
+              ml={
+                valuePercent < 25
+                  ? 0
+                  : valuePercent > 75
+                  ? -valueText.length + "ch"
+                  : -valueText.length / 2 + "ch"
+              }
```
Comment (microbit-robert) [inline src/simulator/RangeSensor.tsx:134]: Open to alternative suggestions to keep the labels from disappearing outside the container.

Comment (microbit-robert) [reply src/simulator/RangeSensor.tsx]: This feels much better [`8c06fd5`](https://github.com/microbit-foundation/python-editor-next/pull/923/commits/8c06fd57656f288b999ab5843ebfb7149eef45d3).

**Review by microbit-matt-hillsdon — APPROVED** [review]
Looks great, thanks.

**Pull request merged**